pkg/server/flags.go

```diff
@@ -0,0 +1,160 @@
+	LogSourceIPsRegex  string `yaml:"log_source_ips_regex"`
+
+	GRPC GRPCFlags `yaml:",inline"`
+	HTTP HTTPFlags `yaml:",inline"`
+}
+
+// HTTPFlags hold static configuration options for the HTTP server.
+type HTTPFlags struct {
+	UseTLS bool `yaml:"-"`
+
+	ListenNetwork string `yaml:"http_listen_network"`
+	ListenAddress string `yaml:"-"` // host:port, takes precedence over ListenHost:ListenPort
+	ListenHost    string `yaml:"http_listen_address"`
+	ListenPort    int    `yaml:"http_listen_port"`
+	ConnLimit     int    `yaml:"http_listen_conn_limit"`
+
+	ReadTimeout  time.Duration `yaml:"http_server_read_timeout"`
+	WriteTimeout time.Duration `yaml:"http_server_write_timeout"`
+	IdleTimeout  time.Duration `yaml:"http_server_idle_timeout"`
+}
+
+// GetListenAddress determines the final ListenAddress, where it is either
+// o.ListenAddress or a combination of o.ListenHost and o.ListenPort.
+func (f HTTPFlags) GetListenAddress() string {
+	if f.ListenAddress != "" {
+		return f.ListenAddress
+	}
+	return fmt.Sprintf("%s:%d", f.ListenHost, f.ListenPort)
+}
+
+// GRPCFlags hold static configuration options for the gRPC server.
+type GRPCFlags struct {
+	UseTLS bool `yaml:"-"`
+
+	ListenNetwork string `yaml:"grpc_listen_network"`
+	ListenAddress string `yaml:"-"` // host:port, takes precedence over ListenHost:ListenPort
+	ListenHost    string `yaml:"grpc_listen_address"`
+	ListenPort    int    `yaml:"grpc_listen_port"`
+	ConnLimit     int    `yaml:"grpc_listen_conn_limit"`
+
+	MaxRecvMsgSize           int           `yaml:"grpc_server_max_recv_msg_size"`
+	MaxSendMsgSize           int           `yaml:"grpc_server_max_send_msg_size"`
+	MaxConcurrentStreams     uint          `yaml:"grpc_server_max_concurrent_streams"`
+	MaxConnectionIdle        time.Duration `yaml:"grpc_server_max_connection_idle"`
+	MaxConnectionAge         time.Duration `yaml:"grpc_server_max_connection_age"`
+	MaxConnectionAgeGrace    time.Duration `yaml:"grpc_server_max_connection_age_grace"`
+	KeepaliveTime            time.Duration `yaml:"grpc_server_keepalive_time"`
+	KeepaliveTimeout         time.Duration `yaml:"grpc_server_keepalive_timeout"`
+	MinTimeBetweenPings      time.Duration `yaml:"grpc_server_min_time_between_pings"`
+	PingWithoutStreamAllowed bool          `yaml:"grpc_server_ping_without_stream_allowed"`
+}
+
+// GetListenAddress determines the final ListenAddress, where it is either
+// o.ListenAddress or a combination of o.ListenHost and o.ListenPort.
+func (f GRPCFlags) GetListenAddress() string {
+	if f.ListenAddress != "" {
+		return f.ListenAddress
+	}
+	return fmt.Sprintf("%s:%d", f.ListenHost, f.ListenPort)
+}
+
+var infinity = time.Duration(math.MaxInt64)
+
+// Default options structs.
+var (
+	DefaultFlags = Flags{
+		RegisterInstrumentation: true,
+		GracefulShutdownTimeout: 30 * time.Second,
+
+		HTTP: DefaultHTTPFlags,
+		GRPC: DefaultGRPCFlags,
+	}
+
+	DefaultHTTPFlags = HTTPFlags{
+		ListenNetwork: "tcp",
+		ListenHost:    "0.0.0.0",
+		ListenPort:    80,
```

Comment:
I think :80 means that you'd need root privileges rights on Linux, right? Which might not be ideal when trying things out.

(I initially commented this on agent-local-config.yaml, but here should be a better place)

Reply:
I would lean towards 12345 since 80 is often high in demand or protected port.

Reply:
should gRPC be 12346 then? (This is also our opportunity to pick a port that stands out and doesn't seem like an example).

Some candidates are:

* `12345` for HTTP / `12346` for gRPC
* `19090` for HTTP (since Prometheus is `9090` / `19091` for gRPC.
* `3001` for HTTP (since Grafana is `3000`) / `3002` for gRPC
* something else?

Reply:
I like the 12345/12346. The closer to our examples the better.

Reply:
Will do 12345/12346 in a follow up PR